src/lightning_lite/lite.py

```diff
@@ -345,6 +345,49 @@ def all_gather(
     def broadcast(self, obj: TBroadcast, src: int = 0) -> TBroadcast:
         return self._strategy.broadcast(obj, src=src)
 
+    @contextmanager
+    def skip_backward_sync(self, module: _LiteModule, enabled: bool = True) -> Generator:
```

Comment:
I'd rename this to `no_backward_sync` or just `no_sync` as done in torch. This would allow use to include this for single device strategies and have a no-op if necessary even though we technically don't skip it. This is also more aligned with torch api as `torch.no_grad`

Reply:
no_backward_sync is fine with me. no_sync is proposed in https://github.com/Lightning-AI/lightning/issues/14713 too but I've stayed away from it because the expression is too general. I think it would be better to have the name specifically point out that this applies to syncing gradients from backward, and nothing else.

Reply:
+1 to `no_backward_sync`

Reply:
@carmocca @justusschock in your preferred choice for no_backward_sync, have you considered the semantics wrt. the enabled=True/False argument. 

One can argue that no_backward_sync(enabled=True) is confusing as it is not clear whether it is doing the sync or not without reading the documentation. It is the same in torch with no_grad, I know. But just so that you are aware of it.

Reply:
I thought about it, but prefer to be consistently inconsistent with what torch does.